apps/dep-graph-client/src/app/graph.ts

```diff
@@ -0,0 +1,384 @@
+            'text-valign': 'center',
+            'padding-left': '16px',
+          },
+        },
+        {
+          selector: 'node[type="e2e"]',
+          style: {
+            shape: 'ellipse',
+            label: 'data(id)',
+            width: 'label',
+            backgroundColor: NrwlPallette.white,
+            'border-style': 'solid',
+            'border-color': NrwlPallette.black,
+            'border-width': '1px',
+            'text-halign': 'center',
+            'text-valign': 'center',
+            'padding-left': '16px',
+          },
+        },
+        {
+          selector: 'node.focused',
+          style: {
+            color: NrwlPallette.twilight,
+            'border-color': NrwlPallette.twilight,
+          },
+        },
+        {
+          selector: 'node.affected',
+          style: {
+            'border-color': NrwlPallette.red,
+          },
+        },
+        {
+          selector: 'edge',
+          style: {
+            width: '1px',
+            'line-color': NrwlPallette.black,
+            'curve-style': 'straight',
+            'target-arrow-shape': 'triangle',
+            'target-arrow-fill': 'filled',
+            'target-arrow-color': NrwlPallette.black,
+          },
+        },
+        {
+          selector: 'edge.affected',
+          style: {
+            'line-color': NrwlPallette.red,
+            'target-arrow-color': NrwlPallette.red,
+          },
+        },
+        {
+          selector: 'edge.implicit',
+          style: {
+            label: 'implicit',
+            'font-size': '8px',
+            'edge-text-rotation': 'autorotate',
+          },
+        },
+        {
+          selector: 'edge.dynamic',
+          style: {
+            'line-dash-pattern': [5, 5],
+            'line-style': 'dashed',
+          },
+        },
+        {
+          selector: ':parent',
+          style: {
+            'background-opacity': 0.5,
+            'background-color': NrwlPallette.twilight,
+            'border-color': NrwlPallette.black,
+            label: 'data(label)',
+            'text-halign': 'center',
+            'text-valign': 'top',
+            'font-weight': 'bold',
+            'font-size': '24px',
+          },
+        },
+      ],
+    });
+
+    this.graph.anywherePanning();
+
+    this.graph.on('zoom', () => {
+      if (this.openTooltip) {
+        this.openTooltip.hide();
+        this.openTooltip = null;
+      }
+    });
+  }
+
+  render() {
+    this.generateCytoscapeLayout();
+    this.addTooltips();
+  }
+
+  addTooltips() {
+    this.graph.$('node:childless').on('click', (event) => {
+      const node = event.target;
+
+      let ref = node.popperRef(); // used only for positioning
+
+      // unfortunately, a dummy element must be passed as tippy only accepts a dom element as the target
+      // https://github.com/atomiks/tippyjs/issues/661
+      let dummyDomEle = document.createElement('div');
+
+      let tip = tippy(dummyDomEle, {
```

Comment:
This is less important.. but maybe even this could move into it's own file.. so that we could do the following:

```
this.graph.$('node:childless').on('click', event => {
  showTooltip(event.target);
})